aws/sdk/integration-tests/s3/tests/timeouts.rs

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_sdk_s3::model::{
+    CompressionType, CsvInput, CsvOutput, ExpressionType, FileHeaderInfo, InputSerialization,
+    OutputSerialization,
+};
+use aws_sdk_s3::{Client, Config, Credentials, Region};
+use aws_smithy_async::assert_elapsed;
+use aws_smithy_async::rt::sleep::{AsyncSleep, TokioSleep};
+use aws_smithy_client::never::NeverService;
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_http::result::ConnectorError;
+use aws_smithy_types::timeout::TimeoutConfig;
+use std::sync::Arc;
+use std::time::Duration;
+
+#[tokio::test]
+async fn test_timeout_service_ends_request_that_never_completes() {
+    let conn: NeverService<http::Request<SdkBody>, http::Response<SdkBody>, ConnectorError> =
+        NeverService::new();
+    let region = Region::from_static("us-east-2");
+    let credentials = Credentials::from_keys("test", "test", None);
+    let timeout_config =
+        TimeoutConfig::new().with_api_call_timeout(Some(Duration::from_secs_f32(0.5)));
+    let sleep_impl: Arc<dyn AsyncSleep> = Arc::new(TokioSleep::new());
+    let config = Config::builder()
+        .region(region)
+        .credentials_provider(credentials)
+        .timeout_config(timeout_config)
+        .sleep_impl(sleep_impl)
+        .build();
+    let client = Client::from_conf_conn(config, conn.clone());
+
+    let now = tokio::time::Instant::now();
+    // When I try to run this test it says there's no `pause` in `tokio::time`
```

Comment:
did you mean to leave this comment in? https://docs.rs/tokio/1.14.0/tokio/time/fn.pause.html (need another cargo feature enabled)